### PR TITLE
Implemented a verification procedure for OssSecurityScore

### LINKED ISSUE #26: Test vectors for score-based scores
Currently it's possible to add test vectors only for ratings. #14 introduces test vectors for feature-based scores. However, we also have scores which consume other sub-scores. For example, see `OssSecurityScore` which is based on `WeightedCompositeScore`. Such scores needs to be covered by test vectors as well.

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here is a list of main updates:

- Added `OssSecurityScore.Verification` class
- Added `OssSecurityScoreTestVectors` notebook which defines the test vectors
- Added `OssSecurityScoreVerification` test which runs the verification
- Updated `loadTestVectorsFromCsvResource()` to expect a set of scores
- Updated `AbstractScore.value()` to return a ScoreValue
- `WeightedCompositeScore` now expects an unknown value for a score
- Fixed checkstyle warnings in the modified files
- Updated README and javadoc

This closes #26

### CODE REVIEW

File: src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerification.java [hunk 1]
```diff
@@ -108,7 +108,7 @@
 
         String label = record.get("label");
         Label expectedLabel = null;
-        if (!"None".equals(label) && !label.isEmpty() && labelParser != null) {
+        if (!"None".equals(label) && !label.isEmpty()) {
```
Comment (Sachpat) [inline src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerification.java]: Will it be a good idea to extract this "None" value to a global constant and use it everywhere?  Just to counter the fact that spelling differences, case sensitivity etc?

Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerification.java]: The constant is currently used only in the `AbstractVerification` class. If we start using this constant anywhere else, we'll definitely extract it to a final static field.

File: src/main/java/com/sap/sgs/phosphor/fosstars/model/score/WeightedCompositeScore.java [hunk 2]
```diff
@@ -159,20 +160,22 @@ public int hashCode() {
...
+    }
+
+    // if the set of values contains a value for the specified score, then return it
+    if (value instanceof ScoreValue) {
+      return (ScoreValue) value;
     }
 
-    // otherwise, try to calculate a value for the score
-    return score.calculate(values);
+    throw new IllegalArgumentException(String.format(
```
Comment (Sachpat) [inline src/main/java/com/sap/sgs/phosphor/fosstars/model/score/WeightedCompositeScore.java]: How about adding @throws in the javadoc for this exception?

Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/model/score/WeightedCompositeScore.java]: Yeah, why not.

File: src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerification.java (hunk 1 above)
Comment (Sachpat) [reply src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerification.java]: Makes sense, how about changing it to equalsIgnoreCase ?

Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerification.java]: Good point.

**Pull request merged**